modules/pel/peltool/src.py

```diff
@@ -0,0 +1,302 @@
+        od = OrderedDict()
+        subsectionID = self.stream.get_int(1)
+        subsectionFlags = self.stream.get_int(1)
+        subsectionWordLength = self.stream.get_int(2)
+        currentLength = 4
+        callouts = []
+        while subsectionWordLength * 4 > currentLength:
+            callout = Callout(self.stream)
+            callouts.append(callout)
+            currentLength += callout.flattenedSize()
+        od["Callout Count"] = len(callouts)
+        calloutJsons = []
+        for callout in callouts:
+            json = OrderedDict()
+            if callout.fruIdentity:
+                json["FRU Type"] = failingComponentType[callout.fruIdentity.flags & 0xf0]
+                json["Priority"] = calloutPriorityValues[callout.priority][1]
+                if len(callout.locationCode) > 0:
+                    json["Location Code"] = callout.locationCode
+                if callout.fruIdentity.flags & Flags.pnSupplied.value:
+                    json["Part Number"] = callout.fruIdentity.pnOrProcedureID
+                if callout.fruIdentity.flags & Flags.maintProcSupplied.value:
+                    json["Procedure"] = callout.fruIdentity.pnOrProcedureID
+                    if callout.fruIdentity.pnOrProcedureID in procedureDesc:
+                        json["Description"] = procedureDesc[callout.fruIdentity.pnOrProcedureID]
+                if callout.fruIdentity.flags & Flags.ccinSupplied.value:
+                    json["CCIN"] = callout.fruIdentity.ccin
+                if callout.fruIdentity.flags & Flags.snSupplied.value:
+                    json["Serial Number"] = callout.fruIdentity.sn
+
+            if callout.pceIdentity:
+                if len(callout.pceIdentity.machineType) > 0:
+                    json["PCE MTMS"] = callout.pceIdentity.machineType + \
+                        "_" + callout.pceIdentity.serialNumber
+                if len(callout.pceIdentity.pceName):
+                    json["PCE Name"] = callout.pceIdentity.pceName
+
+            if callout.mru:
+                mru = OrderedDict()
+                mruId = ""
+                for mru in callout.mru.mrus:
+                    mruId += "%08X" % mru.id + ","
+                json["MRU Id"] = mruId[:-1]
+
+            calloutJsons.append(json)
+        od["Callouts"] = calloutJsons
+        out["Callout Section"] = od
+
+    def paser(self, hexwords: list) -> str:
+        if len(hexwords) < 8:
+            print("The length of the hexwords < 8, exit")
+            exit(1)
+
+        name = self.creatorID.lower() + "src"
+        import importlib
+        cls = importlib.import_module("srcparsers." + name + "." + name)
+        try:
+            return cls.parseSRCToJson(self.asciiString, hexwords[0], hexwords[1], hexwords[2],
+                                      hexwords[3], hexwords[4], hexwords[5], hexwords[6], hexwords[7])
+        except:
+            print("Failed to parse src, , module = " + name)
```

Comment:
you can just print the error message (str(e)) to stderr (i assume that can be done in python) and then skip the src details section